docs/docs/classic-ml/tracking/server-security.mdx

```diff
@@ -0,0 +1,394 @@
+
+:::important Version Requirement
+The security middleware features described in this guide are available in **MLflow 3.5.0 and above only**.
+If you're using an earlier version, you'll need to implement your own security measures via reverse proxy or upgrade to MLflow 3.5.0+.
+:::
+
+Starting from version 3.5.0, MLflow includes built-in security middleware that protects your tracking server against common web vulnerabilities. This guide explains the security features, how to configure them, and best practices for deployment.
+
+## Why Security Middleware?
+
+MLflow tracking servers are often exposed to networks beyond localhost, making them potential targets for various attacks. The security middleware provides protection against:
+
+- **DNS Rebinding Attacks**: Prevents attackers from tricking browsers into accessing internal services
+- **Cross-Origin Resource Sharing (CORS) Attacks**: Blocks unauthorized cross-origin requests that could steal data
+- **Host Header Injection**: Prevents attacks that manipulate the Host header for cache poisoning or password reset attacks
+- **Clickjacking**: Prevents embedding MLflow UI in malicious iframes
+- **MIME Type Confusion**: Blocks browsers from incorrectly interpreting file types
+
+## Default Security Behavior
+
+By default, MLflow server is configured for **secure local and internal network development**:
+
+- Accepts connections from localhost addresses (127.0.0.1, ::1, localhost) on any port, and private IP ranges (192.168._, 10._, 172.16-172.31.\*)
+- Allows cross-origin requests from any localhost port (e.g., http://localhost:3000, http://localhost:8080, etc.)
+- Validates Host headers to prevent DNS rebinding attacks
+- Adds security headers to all responses (X-Frame-Options, X-Content-Type-Options)
+- Blocks cross-origin state-changing requests (POST, PUT, DELETE, PATCH) from non-localhost origins unless explicitly configured
+
+This means you can run `mlflow server` on your local machine or internal network without any configuration, and team members can access it from private IP addresses.
+
+## Configuration Options
+
+Security can be configured via CLI options or environment variables. CLI options take precedence over environment variables.
+
+### Security Configuration
+
+| CLI Option               | Environment Variable            | Description                                  | Default                       |
+| ------------------------ | ------------------------------- | -------------------------------------------- | ----------------------------- |
+| `--cors-allowed-origins` | `MLFLOW_CORS_ALLOWED_ORIGINS`   | Comma-separated list of allowed CORS origins | localhost on any port         |
+| `--allowed-hosts`        | `MLFLOW_ALLOWED_HOSTS`          | Comma-separated list of allowed Host headers | localhost + private IP ranges |
+| `--no-host-validation`   | `MLFLOW_HOST_HEADER_VALIDATION` | Disable Host header validation               | Validation enabled            |
```

Comment:
Can we use `MLFLOW_ALLOWED_HOSTS="*"` instead of `MLFLOW_HOST_HEADER_VALIDATION`